app/javascript/application.js

```diff
@@ -0,0 +1,55 @@
+import { initAll } from 'govuk-frontend'
+import Rails from '@rails/ujs'
+
+// require polyfills via core-js
+import 'core-js/stable'
+
+import axios from 'axios'
+
+import './src/modules/proceedings'
+import './src/admin-delete'
+import './src/bank_transactions'
+import './src/checkbox_control'
+import './src/cookie_banner'
+import './src/details-polyfill'
+import './src/dropzone'
+import './src/file-upload-validation'
+import './src/helpers'
+import './src/modal-dialogue'
+import './src/no_script'
+import './src/print_button'
+import './src/start_button'
+import './src/table-select-all'
+import './src/table-sort'
+import './src/worker_waiter'
+
+initAll()
+Rails.start()
+
+axios.interceptors.request.use(function (config) {
+  if (process.env.NODE_ENV === 'test') {
+    const requestCount = document.getElementById('requestCount')
+
+    if (requestCount) {
+      const requestsOutstanding = parseInt(requestCount.dataset.requestsOutstanding)
+      requestCount.dataset.requestsOutstanding = (requestsOutstanding + 1).toString()
+    }
+  }
+  return config
+}, function (error) {
+  return Promise.reject(error)
+})
+
+axios.interceptors.response.use(function (response) {
+  if (process.env.NODE_ENV === 'test') {
+    const requestCount = document.getElementById('requestCount')
+
+    if (requestCount) {
+      const requestsOutstanding = parseInt(requestCount.dataset.requestsOutstanding)
+      requestCount.dataset.requestsOutstanding = (requestsOutstanding - 1).toString()
+    }
+  }
+  return response
+}, function (error) {
+  return Promise.reject(error)
+})
```

Comment:
Before, we relied on `jquery` to tell us if there were any ongoing AJAX requests
in tests, so could wait for them to resolve before making assertions against
asynchronously rendered content.

We shouldn't need to do this, and Capybara should wait appropriately. We might
want to revisit the tests that make use of the `wait_for_ajax` helper to ensure
they are not making network requests, and that Capybara selectors wait long
enough.

Removing jquery means we cannot rely on `$.active` or `AJAX.activeRequestCount`.
Since we are using `axios`, we can configure some crude
[interceptors](https://github.com/axios/axios#interceptors) that increment and
decrement a data attribute on an empty `div` to determine whether requests are
ongoing in development mode.

I would expect us to discuss and iterate on this approach. Alternatives might be
to rethink how our tests work (i.e we expect rendered content or certain
behvaiour after the request), or to use a different driver (e.g Cuprite) that
allows us to easily wait for network idle in our tests.

Reply:
Refactoring the applications' Javascript is part of the larger objectives:

1. Reduce dependencies
2. Use core JS (fetch API)
3. Review JS testing strategy

This would mean not relying on jquery, axios etc.

Reply:
We decided to leave `jQuery` in for now, and we will revisit this as part of a wider JS-focused review and refactor.